From: Lucas Stach <l.stach@pengutronix.de>
Date: Fri, 1 Sep 2017 17:52:18 +0200
Subject: [PATCH] of: populate clock providers before other devices

Clocks are a basic resource, which may be needed early by other
devices or even the bus driver (as is the case with the ARM AMBA
bus). Register them before populating other devices.

Signed-off-by: Lucas Stach <l.stach@pengutronix.de>
---
 drivers/of/base.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/of/base.c b/drivers/of/base.c
index 95bea4ee836d..fb4d2c03946e 100644
--- a/drivers/of/base.c
+++ b/drivers/of/base.c
@@ -1938,8 +1938,8 @@ int of_probe(void)
 	if (memory)
 		of_add_memory(memory, false);
 
-	of_platform_populate(root_node, of_default_bus_match_table, NULL);
 	of_clk_init(root_node, NULL);
+	of_platform_populate(root_node, of_default_bus_match_table, NULL);
 
 	return 0;
 }
